File: career-plan-generator v2.0-PUB/scripts/build_docx.js
#!/usr/bin/env node
/**
 * career-plan-generator · docx 构建脚本
 * --------------------------------------------------
 * 用法：
 *   node build_docx.js plan_data.json output.docx
 *
 * plan_data.json 的结构（见文末 EXAMPLE_DATA 演示）：
 *   {
 *     "studentName": "张三",         // 可选，用于封面 / 页眉 / 收尾
 *     "subtitle": "—— 计算机专业 · 互联网大厂方向 ——",
 *     "epigraph": "...",            // 一句话引言，可空
 *     "generatedAt": "2026-05",     // 拟定日期
 *     "planRange": "2026 — 2030",
 *     "portrait": "200 字以内的学生画像速写",
 *     "chapters": [
 *       {
 *         "title": "第一章 个人现状盘点",
 *         "blocks": [ {type:"p", text:"..."},
 *                     {type:"h2", text:"..."},
 *                     {type:"h3", text:"..."},
 *                     {type:"bullet", items:["...","..."]},
 *                     {type:"numbered", items:["...","..."]},
 *                     {type:"quote", text:"..."},
 *                     {type:"table", header:["列1","列2"], rows:[["..","..."]]},
 *                     {type:"pagebreak"}
 *                   ]
 *       },
 *       ...
 *     ],
 *     "closingLetter": "200 字给学生的一段话"   // 7.E
 *   }
 *
 * 依赖：docx (npm install -g docx)
 * --------------------------------------------------
 */

const fs = require("fs");
const path = require("path");

let docx;
try {
  docx = require("docx");
} catch (e) {
  console.error("[build_docx] 缺少依赖：docx。请先运行 `npm install -g docx`。");
  process.exit(2);
}

const {
  Document, Packer, Paragraph, TextRun, Table, TableRow, TableCell,
  Header, Footer, AlignmentType, LevelFormat, TabStopType, TabStopPosition,
  HeadingLevel, BorderStyle, WidthType, ShadingType,
  VerticalAlign, PageNumber, PageBreak
} = docx;

// ---------- 配置 ----------
const FONT = "Microsoft YaHei";
const COLOR = {
  H1: "1F4E79",
  H2: "2E75B6",
  H3: "404040",
  text: "262626",
  mute: "8C8C8C",
  rule: "B4C7E7",
  tableHead: "D9E2F3",
  quoteBg: "F2F7FB",
  quoteBar: "2E75B6",
};

// ---------- 元素构造器 ----------
function P(text, { bold = false, size = 22, align, color = COLOR.text, italics = false, spacingAfter = 80 } = {}) {
  return new Paragraph({
    alignment: align,
    spacing: { before: 0, after: spacingAfter, line: 360 },
    children: [new TextRun({ text, bold, size, font: FONT, color, italics })],
  });
}

function H1(text) {
  return new Paragraph({
    heading: HeadingLevel.HEADING_1,
    spacing: { before: 360, after: 200 },
    children: [new TextRun({ text, bold: true, size: 36, font: FONT, color: COLOR.H1 })],
  });
}
function H2(text) {
  return new Paragraph({
    heading: HeadingLevel.HEADING_2,
    spacing: { before: 280, after: 160 },
    children: [new TextRun({ text, bold: true, size: 28, font: FONT, color: COLOR.H2 })],
  });
}
function H3(text) {
  return new Paragraph({
    heading: HeadingLevel.HEADING_3,
    spacing: { before: 220, after: 120 },
    children: [new TextRun({ text, bold: true, size: 24, font: FONT, color: COLOR.H3 })],
  });
}

function bulletItem(text) {
  return new Paragraph({
    numbering: { reference: "bullets", level: 0 },
    spacing: { before: 40, after: 40, line: 340 },
    children: [new TextRun({ text, size: 22, font: FONT, color: COLOR.text })],
  });
}
function numberedItem(text) {
  return new Paragraph({
    numbering: { reference: "numbers", level: 0 },
    spacing: { before: 40, after: 40, line: 340 },
    children: [new TextRun({ text, size: 22, font: FONT, color: COLOR.text })],
  });
}

function quoteBlock(text) {
  return new Paragraph({
    spacing: { before: 120, after: 120, line: 360 },
    indent: { left: 360, right: 360 },
    border: { left: { style: BorderStyle.SINGLE, size: 24, color: COLOR.quoteBar, space: 8 } },
    shading: { fill: COLOR.quoteBg, type: ShadingType.CLEAR },
    children: [new TextRun({ text, italics: true, size: 22, font: FONT, color: COLOR.H3 })],
  });
}

const cellBorder = { style: BorderStyle.SINGLE, size: 4, color: COLOR.rule };
const cellBorders = { top: cellBorder, bottom: cellBorder, left: cellBorder, right: cellBorder };

function cell(text, { bold = false, fill, width, align } = {}) {
  return new TableCell({
    borders: cellBorders,
    width: width ? { size: width, type: WidthType.DXA } : undefined,
    shading: fill ? { fill, type: ShadingType.CLEAR } : undefined,
    margins: { top: 100, bottom: 100, left: 140, right: 140 },
    verticalAlign: VerticalAlign.CENTER,
    children: [new Paragraph({
      alignment: align,
      children: [new TextRun({ text: String(text ?? ""), bold, size: 20, font: FONT, color: COLOR.text })],
    })],
  });
}

function buildTable(header, rows) {
  // 默认 A4 1英寸边距下内容宽度约 9026 DXA，平均分配列宽
  const totalWidth = 9026;
  const colCount = header.length;
  const colWidth = Math.floor(totalWidth / colCount);
  const columnWidths = Array(colCount).fill(colWidth);

  const headerRow = new TableRow({
    tableHeader: true,
    children: header.map(h => cell(h, { bold: true, fill: COLOR.tableHead, width: colWidth, align: AlignmentType.CENTER })),
  });
  const dataRows = rows.map(row => new TableRow({
    children: row.map(c => cell(c, { width: colWidth })),
  }));

  return new Table({
    width: { size: totalWidth, type: WidthType.DXA },
    columnWidths,
    rows: [headerRow, ...dataRows],
  });
}

// ---------- block 渲染分发器 ----------
function renderBlock(block) {
  switch (block.type) {
    case "p":         return [P(block.text, block.opts || {})];
    case "h1":        return [H1(block.text)];
    case "h2":        return [H2(block.text)];
    case "h3":        return [H3(block.text)];
    case "bullet":    return (block.items || []).map(bulletItem);
    case "numbered":  return (block.items || []).map(numberedItem);
    case "quote":     return [quoteBlock(block.text)];
    case "table":     return [buildTable(block.header || [], block.rows || []), P(" ")];
    case "pagebreak": return [new Paragraph({ children: [new PageBreak()] })];
    case "blank":     return [P(" ")];
    default:
      return [P("[未知块类型：" + block.type + "]", { color: "C00000" })];
  }
}

// ---------- 主文档构造 ----------
function buildDoc(planData) {
  const studentName  = planData.studentName  || "同学";
  const subtitle     = planData.subtitle     || "—— 定制版职业规划 ——";
  const epigraph     = planData.epigraph     || "";
  const generatedAt  = planData.generatedAt  || "";
  const planRange    = planData.planRange    || "";
  const portrait     = planData.portrait     || "";
  const closingLetter = planData.closingLetter || "";

  const children = [];

  // 封面
  children.push(
    new Paragraph({ spacing: { before: 1800, after: 200 }, alignment: AlignmentType.CENTER,
      children: [new TextRun({ text: "职业规划成长计划书", bold: true, size: 56, font: FONT, color: COLOR.H1 })] }),
    new Paragraph({ spacing: { before: 200, after: 200 }, alignment: AlignmentType.CENTER,
      children: [new TextRun({ text: subtitle, size: 28, font: FONT, color: COLOR.H2 })] }),
  );
  if (epigraph) {
    children.push(
      new Paragraph({ spacing: { before: 1200, after: 100 }, alignment: AlignmentType.CENTER,
        children: [new TextRun({ text: `「${epigraph}」`, italics: true, size: 24, font: FONT, color: "595959" })] })
    );
  }
  children.push(
    new Paragraph({ spacing: { before: 1400, after: 80 }, alignment: AlignmentType.CENTER,
      children: [new TextRun({ text: `署名：${studentName}`, size: 24, font: FONT, color: COLOR.text })] }),
  );
  if (planRange) children.push(
    new Paragraph({ spacing: { before: 80, after: 80 }, alignment: AlignmentType.CENTER,
      children: [new TextRun({ text: `规划周期：${planRange}`, size: 24, font: FONT, color: COLOR.text })] }),
  );
  if (generatedAt) children.push(
    new Paragraph({ spacing: { before: 80, after: 80 }, alignment: AlignmentType.CENTER,
      children: [new TextRun({ text: `拟定日期：${generatedAt}`, size: 24, font: FONT, color: COLOR.text })] }),
  );
  // v1.4 新增：封面底部声明小字
  children.push(
    new Paragraph({ spacing: { before: 2400, after: 60 }, alignment: AlignmentType.CENTER,
      children: [new TextRun({
        text: "本报告由国科职规体系 + AI 生成，受限于模型能力，仅供参看。",
        size: 16, font: FONT, color: "8C8C8C", italics: true
      })] }),
    new Paragraph({ spacing: { before: 20, after: 80 }, alignment: AlignmentType.CENTER,
      children: [new TextRun({
        text: "如有需求，可向国科职规老师获取专业能力测评与职规服务。",
        size: 16, font: FONT, color: "8C8C8C", italics: true
      })] }),
  );
  children.push(new Paragraph({ children: [new PageBreak()] }));

  // 学生画像速写
  if (portrait) {
    children.push(
      H1("学生画像速写"),
      quoteBlock(portrait),
      new Paragraph({ children: [new PageBreak()] })
    );
  }

  // 各章
  for (const chapter of (planData.chapters || [])) {
    children.push(H1(chapter.title || ""));
    for (const block of (chapter.blocks || [])) {
      for (const node of renderBlock(block)) children.push(node);
    }
    children.push(new Paragraph({ children: [new PageBreak()] }));
  }

  // 收尾那段话
  if (closingLetter) {
    children.push(
      H1("写给你的一封信"),
      quoteBlock(closingLetter),
      P(" "),
      P("—— 本计划书 v1.0 完 ——", { align: AlignmentType.CENTER, color: COLOR.mute, italics: true })
    );
  }

  const doc = new Document({
    creator: "career-plan-generator",
    title: `${studentName} - 职业规划成长计划书`,
    styles: {
      default: { document: { run: { font: FONT, size: 22 } } },
      paragraphStyles: [
        { id: "Heading1", name: "Heading 1", basedOn: "Normal", next: "Normal", quickFormat: true,
          run: { size: 36, bold: true, font: FONT, color: COLOR.H1 },
          paragraph: { spacing: { before: 360, after: 200 }, outlineLevel: 0 } },
        { id: "Heading2", name: "Heading 2", basedOn: "Normal", next: "Normal", quickFormat: true,
          run: { size: 28, bold: true, font: FONT, color: COLOR.H2 },
          paragraph: { spacing: { before: 280, after: 160 }, outlineLevel: 1 } },
        { id: "Heading3", name: "Heading 3", basedOn: "Normal", next: "Normal", quickFormat: true,
          run: { size: 24, bold: true, font: FONT, color: COLOR.H3 },
          paragraph: { spacing: { before: 220, after: 120 }, outlineLevel: 2 } },
      ],
    },
    numbering: {
      config: [
        { reference: "bullets",
          levels: [{ level: 0, format: LevelFormat.BULLET, text: "•", alignment: AlignmentType.LEFT,
            style: { paragraph: { indent: { left: 720, hanging: 360 } } } }] },
        { reference: "numbers",
          levels: [{ level: 0, format: LevelFormat.DECIMAL, text: "%1.", alignment: AlignmentType.LEFT,
            style: { paragraph: { indent: { left: 720, hanging: 360 } } } }] },
      ],
    },
    sections: [{
      properties: {
        page: {
          size: { width: 11906, height: 16838 }, // A4
          margin: { top: 1440, right: 1440, bottom: 1440, left: 1440 },
        },
      },
      headers: {
        default: new Header({ children: [new Paragraph({
          alignment: AlignmentType.RIGHT,
          children: [new TextRun({ text: `${studentName}的职业规划成长计划书 · v1.4`, size: 18, font: FONT, color: COLOR.mute })],
        })] }),
      },
      footers: {
        default: new Footer({ children: [new Paragraph({
          alignment: AlignmentType.CENTER,
          children: [
            new TextRun({ text: "— 第 ", size: 18, font: FONT, color: COLOR.mute }),
            new TextRun({ children: [PageNumber.CURRENT], size: 18, font: FONT, color: COLOR.mute }),
            new TextRun({ text: " 页 —", size: 18, font: FONT, color: COLOR.mute }),
          ],
        })] }),
      },
      children,
    }],
  });

  return doc;
}

// ---------- CLI 入口 ----------
async function main() {
  const [, , dataPath, outPath] = process.argv;
  if (!dataPath || !outPath) {
    console.error("用法：node build_docx.js <plan_data.json> <output.docx>");
    process.exit(1);
  }
  const raw = fs.readFileSync(dataPath, "utf-8");
  const planData = JSON.parse(raw);
  const doc = buildDoc(planData);
  const buffer = await Packer.toBuffer(doc);
  fs.writeFileSync(outPath, buffer);
  console.log(`[build_docx] 已生成：${outPath}（${buffer.length} 字节）`);
}

// 示例数据（仅用于参考结构，不会被自动执行）
const EXAMPLE_DATA = {
  studentName: "张三",
  subtitle: "—— 计算机专业 · 互联网大厂方向 ——",
  epigraph: "凡事预则立，不预则废。",
  generatedAt: "2026-05",
  planRange: "2026 — 2030",
  portrait: "（200 字以内的学生画像速写……）",
  chapters: [
    {
      title: "第一章 个人现状盘点",
      blocks: [
        { type: "h2", text: "1.1 学业表现" },
        { type: "p",  text: "（具体到学生填写的绩点 / 排名…）" },
        { type: "bullet", items: ["绩点 3.85 / 4.0", "专业排名前 15%"] },
        { type: "table",
          header: ["维度", "现状", "评估"],
          rows: [["GPA", "3.85", "稳"], ["排名", "前 15%", "强"]] },
      ],
    },
  ],
  closingLetter: "（200 字给学生的一段话……）",
};

if (require.main === module) {
  main().catch(err => { console.error(err); process.exit(1); });
}

module.exports = { buildDoc, EXAMPLE_DATA };

File: career-plan-generator v2.0-PUB/scripts/build_docx.py
#!/usr/bin/env python3
"""
career-plan-generator · docx 备份生成器（Python 版）

用法:
    python build_docx.py plan_data.json output.docx

依赖:
    pip install python-docx

设计目的:
    保证「在任何 Agent / 任何模型环境下都能生成 .docx 文件」。
    优先顺序：node + docx-js（首选）→ Python + python-docx（本脚本）→ Pandoc → 最后才退化为 md。

数据结构（与 build_docx.js 完全兼容）:
    {
      "studentName": "称呼",
      "subtitle": "—— 子标题 ——",
      "epigraph": "可选引言",
      "generatedAt": "2026-05",
      "planRange": "2026 — 2030",
      "portrait": "学生画像速写",
      "chapters": [
        {
          "title": "第X章 ...",
          "blocks": [
            {"type": "p", "text": "段落..."},
            {"type": "h2", "text": "二级标题"},
            {"type": "h3", "text": "三级标题"},
            {"type": "bullet", "items": ["...", "..."]},
            {"type": "numbered", "items": ["...", "..."]},
            {"type": "quote", "text": "引用块"},
            {"type": "table", "header": ["列1", "列2"], "rows": [["a","b"]]},
            {"type": "pagebreak"},
            {"type": "blank"}
          ]
        }
      ],
      "closingLetter": "写给你的一封信"
    }
"""

import json
import sys
from datetime import date

try:
    from docx import Document
    from docx.shared import Pt, RGBColor, Cm, Inches
    from docx.enum.text import WD_ALIGN_PARAGRAPH, WD_BREAK
    from docx.enum.table import WD_ALIGN_VERTICAL
    from docx.oxml.ns import qn
    from docx.oxml import OxmlElement
except ImportError:
    print("[build_docx.py] 缺少依赖：python-docx。请先运行 `pip install python-docx`。", file=sys.stderr)
    sys.exit(2)


# ---------- 配置 ----------
FONT_CN = "Microsoft YaHei"
FONT_EN = "Calibri"

COLOR = {
    "H1": RGBColor(0x1F, 0x4E, 0x79),
    "H2": RGBColor(0x2E, 0x75, 0xB6),
    "H3": RGBColor(0x40, 0x40, 0x40),
    "text": RGBColor(0x26, 0x26, 0x26),
    "mute": RGBColor(0x8C, 0x8C, 0x8C),
    "rule": RGBColor(0xB4, 0xC7, 0xE7),
    "table_head": RGBColor(0xD9, 0xE2, 0xF3),
    "quote_bg": "F2F7FB",
    "quote_bar": RGBColor(0x2E, 0x75, 0xB6),
}


# ---------- 辅助 ----------
def set_run_font(run, font=FONT_CN, size=11, bold=False, italic=False, color=None):
    """对一个 run 设置字体（含中文东亚字体）"""
    run.font.name = FONT_EN
    run.font.size = Pt(size)
    run.font.bold = bold
    run.font.italic = italic
    if color is not None:
        run.font.color.rgb = color
    # 中文字体设置
    rPr = run._element.get_or_add_rPr()
    rFonts = rPr.find(qn("w:rFonts"))
    if rFonts is None:
        rFonts = OxmlElement("w:rFonts")
        rPr.append(rFonts)
    rFonts.set(qn("w:eastAsia"), font)
    rFonts.set(qn("w:hAnsi"), FONT_EN)
    rFonts.set(qn("w:ascii"), FONT_EN)


def add_paragraph(doc, text, *, size=11, bold=False, italic=False,
                  color=None, align=None, space_before=0, space_after=4):
    p = doc.add_paragraph()
    if align == "center":
        p.alignment = WD_ALIGN_PARAGRAPH.CENTER
    elif align == "right":
        p.alignment = WD_ALIGN_PARAGRAPH.RIGHT
    p.paragraph_format.space_before = Pt(space_before)
    p.paragraph_format.space_after = Pt(space_after)
    p.paragraph_format.line_spacing = 1.5
    run = p.add_run(text)
    set_run_font(run, size=size, bold=bold, italic=italic, color=color or COLOR["text"])
    return p


def add_heading_1(doc, text):
    p = doc.add_paragraph()
    p.paragraph_format.space_before = Pt(18)
    p.paragraph_format.space_after = Pt(10)
    run = p.add_run(text)
    set_run_font(run, size=20, bold=True, color=COLOR["H1"])
    return p


def add_heading_2(doc, text):
    p = doc.add_paragraph()
    p.paragraph_format.space_before = Pt(14)
    p.paragraph_format.space_after = Pt(8)
    run = p.add_run(text)
    set_run_font(run, size=15, bold=True, color=COLOR["H2"])
    return p


def add_heading_3(doc, text):
    p = doc.add_paragraph()
    p.paragraph_format.space_before = Pt(10)
    p.paragraph_format.space_after = Pt(6)
    run = p.add_run(text)
    set_run_font(run, size=13, bold=True, color=COLOR["H3"])
    return p


def add_bullet(doc, text):
    p = doc.add_paragraph(style="List Bullet")
    p.paragraph_format.space_before = Pt(2)
    p.paragraph_format.space_after = Pt(2)
    p.paragraph_format.line_spacing = 1.4
    run = p.add_run(text)
    set_run_font(run, size=11, color=COLOR["text"])
    return p


def add_numbered(doc, text):
    p = doc.add_paragraph(style="List Number")
    p.paragraph_format.space_before = Pt(2)
    p.paragraph_format.space_after = Pt(2)
    p.paragraph_format.line_spacing = 1.4
    run = p.add_run(text)
    set_run_font(run, size=11, color=COLOR["text"])
    return p


def add_quote(doc, text):
    p = doc.add_paragraph()
    p.paragraph_format.left_indent = Cm(0.6)
    p.paragraph_format.right_indent = Cm(0.6)
    p.paragraph_format.space_before = Pt(6)
    p.paragraph_format.space_after = Pt(6)
    # 设置背景色 + 左边框
    pPr = p._element.get_or_add_pPr()
    shd = OxmlElement("w:shd")
    shd.set(qn("w:val"), "clear")
    shd.set(qn("w:color"), "auto")
    shd.set(qn("w:fill"), COLOR["quote_bg"])
    pPr.append(shd)
    pBdr = OxmlElement("w:pBdr")
    left = OxmlElement("w:left")
    left.set(qn("w:val"), "single")
    left.set(qn("w:sz"), "24")
    left.set(qn("w:space"), "8")
    left.set(qn("w:color"), "2E75B6")
    pBdr.append(left)
    pPr.append(pBdr)
    run = p.add_run(text)
    set_run_font(run, size=11, italic=True, color=COLOR["H3"])
    return p


def add_table(doc, header, rows):
    table = doc.add_table(rows=1 + len(rows), cols=len(header))
    table.style = "Light Grid Accent 1"
    table.alignment = WD_ALIGN_PARAGRAPH.CENTER
    # 表头
    hdr = table.rows[0].cells
    for i, h in enumerate(header):
        hdr[i].text = ""
        p = hdr[i].paragraphs[0]
        p.alignment = WD_ALIGN_PARAGRAPH.CENTER
        run = p.add_run(str(h))
        set_run_font(run, size=10, bold=True, color=COLOR["text"])
        # 表头背景
        tcPr = hdr[i]._tc.get_or_add_tcPr()
        shd = OxmlElement("w:shd")
        shd.set(qn("w:val"), "clear")
        shd.set(qn("w:fill"), "D9E2F3")
        tcPr.append(shd)
    # 数据行
    for ri, row in enumerate(rows, start=1):
        cells = table.rows[ri].cells
        for ci, val in enumerate(row):
            cells[ci].text = ""
            p = cells[ci].paragraphs[0]
            run = p.add_run(str(val) if val is not None else "")
            set_run_font(run, size=10, color=COLOR["text"])
    return table


def add_page_break(doc):
    p = doc.add_paragraph()
    run = p.add_run()
    run.add_break(WD_BREAK.PAGE)


# ---------- 块渲染分发 ----------
def render_block(doc, block):
    t = block.get("type", "")
    if t == "p":
        opts = block.get("opts", {}) or {}
        add_paragraph(
            doc, block.get("text", ""),
            size=opts.get("size", 11),
            bold=opts.get("bold", False),
            italic=opts.get("italics", False),
            align=opts.get("align"),
            color=COLOR.get(opts.get("color"), None) if isinstance(opts.get("color"), str) else None,
        )
    elif t == "h1":
        add_heading_1(doc, block.get("text", ""))
    elif t == "h2":
        add_heading_2(doc, block.get("text", ""))
    elif t == "h3":
        add_heading_3(doc, block.get("text", ""))
    elif t == "bullet":
        for item in block.get("items", []):
            add_bullet(doc, item)
    elif t == "numbered":
        for item in block.get("items", []):
            add_numbered(doc, item)
    elif t == "quote":
        add_quote(doc, block.get("text", ""))
    elif t == "table":
        add_table(doc, block.get("header", []), block.get("rows", []))
        add_paragraph(doc, " ")
    elif t == "pagebreak":
        add_page_break(doc)
    elif t == "blank":
        add_paragraph(doc, " ")
    else:
        add_paragraph(doc, f"[未知块类型：{t}]", color=RGBColor(0xC0, 0x00, 0x00))


# ---------- 主构造 ----------
def build(plan_data, out_path):
    doc = Document()

    # A4 + 边距
    for section in doc.sections:
        section.page_width = Cm(21.0)
        section.page_height = Cm(29.7)
        section.top_margin = Inches(1)
        section.bottom_margin = Inches(1)
        section.left_margin = Inches(1)
        section.right_margin = Inches(1)

    # 页眉
    student_name = plan_data.get("studentName") or "你"
    header = doc.sections[0].header
    hp = header.paragraphs[0]
    hp.alignment = WD_ALIGN_PARAGRAPH.RIGHT
    hr = hp.add_run(f"{student_name}的职业规划成长计划书 · v1.4")
    set_run_font(hr, size=9, color=COLOR["mute"])

    # 页脚（页码）
    footer = doc.sections[0].footer
    fp = footer.paragraphs[0]
    fp.alignment = WD_ALIGN_PARAGRAPH.CENTER
    fr = fp.add_run("— 第 ")
    set_run_font(fr, size=9, color=COLOR["mute"])
    # 页码字段
    field = OxmlElement("w:fldSimple")
    field.set(qn("w:instr"), "PAGE")
    fp._p.append(field)
    fr2 = fp.add_run(" 页 —")
    set_run_font(fr2, size=9, color=COLOR["mute"])

    # ========== 封面 ==========
    add_paragraph(doc, "", space_before=80)
    p = doc.add_paragraph()
    p.alignment = WD_ALIGN_PARAGRAPH.CENTER
    p.paragraph_format.space_before = Pt(80)
    p.paragraph_format.space_after = Pt(10)
    run = p.add_run("职业规划成长计划书")
    set_run_font(run, size=28, bold=True, color=COLOR["H1"])

    subtitle = plan_data.get("subtitle") or "—— 定制版职业规划 ——"
    p = doc.add_paragraph()
    p.alignment = WD_ALIGN_PARAGRAPH.CENTER
    p.paragraph_format.space_before = Pt(8)
    p.paragraph_format.space_after = Pt(60)
    run = p.add_run(subtitle)
    set_run_font(run, size=14, color=COLOR["H2"])

    epigraph = plan_data.get("epigraph")
    if epigraph:
        p = doc.add_paragraph()
        p.alignment = WD_ALIGN_PARAGRAPH.CENTER
        p.paragraph_format.space_before = Pt(60)
        p.paragraph_format.space_after = Pt(4)
        run = p.add_run(f"「{epigraph}」")
        set_run_font(run, size=12, italic=True, color=RGBColor(0x59, 0x59, 0x59))

    add_paragraph(doc, " ", space_before=60)
    add_paragraph(doc, f"署名：{student_name}", size=12, align="center", space_before=4, space_after=4)

    plan_range = plan_data.get("planRange")
    if plan_range:
        add_paragraph(doc, f"规划周期：{plan_range}", size=12, align="center", space_before=4, space_after=4)
    generated_at = plan_data.get("generatedAt")
    if generated_at:
        add_paragraph(doc, f"拟定日期：{generated_at}", size=12, align="center", space_before=4, space_after=4)

    # v1.4 封面底部国科声明
    add_paragraph(doc, " ", space_before=80)
    add_paragraph(
        doc, "本报告由国科职规体系 + AI 生成，受限于模型能力，仅供参看。",
        size=8, italic=True, align="center", color=COLOR["mute"], space_before=80, space_after=2
    )
    add_paragraph(
        doc, "如有需求，可向国科职规老师获取专业能力测评与职规服务。",
        size=8, italic=True, align="center", color=COLOR["mute"], space_before=2, space_after=4
    )

    add_page_break(doc)

    # ========== 学生画像速写 ==========
    portrait = plan_data.get("portrait")
    if portrait:
        add_heading_1(doc, "学生画像速写")
        add_quote(doc, portrait)
        add_page_break(doc)

    # ========== 各章 ==========
    for chapter in plan_data.get("chapters", []):
        add_heading_1(doc, chapter.get("title", ""))
        for block in chapter.get("blocks", []):
            render_block(doc, block)
        add_page_break(doc)

    # ========== 第十三章 · 5. 写给你的一封信 ==========
    closing = plan_data.get("closingLetter")
    if closing:
        add_heading_1(doc, "写给你的一封信")
        add_quote(doc, closing)
        add_paragraph(doc, " ")
        add_paragraph(
            doc, "—— 本计划书 v1.4 完 ——",
            italic=True, align="center", color=COLOR["mute"], space_before=12, space_after=4
        )

    doc.save(out_path)
    print(f"[build_docx.py] 已生成：{out_path}")


def main():
    if len(sys.argv) < 3:
        print("用法：python build_docx.py <plan_data.json> <output.docx>", file=sys.stderr)
        sys.exit(1)
    with open(sys.argv[1], "r", encoding="utf-8") as f:
        plan_data = json.load(f)
    build(plan_data, sys.argv[2])


if __name__ == "__main__":
    main()
